libs/ktem/ktem/tags/crud.py

```diff
@@ -137,26 +125,25 @@ class ChunkTagIndexCRUD:
     def __init__(self, engine):
         self._engine = engine
 
-    @property
+    @cached_property
     def tag_crud(self) -> TagCRUD:
         return TagCRUD(self._engine)
 
-    def list_all(self) -> list[dict]:
+    def list_all(self) -> list[ChunkTagIndex]:
         with Session(self._engine) as session:
             statement = select(ChunkTagIndex)
             results = session.exec(statement).all()
 
-            results = [dict(result) for result in results]
         return results
 
-    def query_by_id(self, record_id: str) -> dict | None:
+    def query_by_id(self, record_id: str) -> ChunkTagIndex | None:
         with Session(self._engine) as session:
             statement = select(ChunkTagIndex).where(ChunkTagIndex.id == record_id)
 
             result = session.exec(statement).first()
             return result
 
-    def query_by_chunk_ids(self, chunk_ids: list[str]) -> dict[str, dict]:
+    def query_by_chunk_ids(self, chunk_ids: list[str]) -> dict[str, dict[str, str]]:

```

Comment:
since the return value is `chunk_id_to_tag`, the expected type should be `dict[str, Tag]` right? no motivation to output `dict[str, str]`